### DISCUSSION TITLE
Integrating Tesla Online

### DISCUSSION THREAD

**Discussion by ZenGuy99** [Kemmey/Kemmey-TeslaWatch-Public#495] [Ideas]
I love Watch for Tesla and use it everyday in conjunction with Apple Watch. The most frustrating challenge is the “state” the car is in when the open charge port command is issued. More often than not I’m told the car is off line. I use to open and close a door just to wake the car, but wonder if it’s possible to incorporate a wake function into commands? It would be awesome to execute the commands via Siri as I head toward the garage so by the time I was there the car would wake and open the charge port… same thing for unlock chargers port. 

Fingers crossed 🤞🏻

**Comment by Kemmey** [comment]
Hey - this is actually already the case. Due to some technical limitations, the Watch app only has about 15 seconds of background execution time, so it's not always possible to wake the car in that time. I have a workaround for that, where Siri is supposed to ask "keep waiting for wake?" - but on watchOS 9 that workaround seems to be somewhat broken. I'll see what I can do to fix that. But ideally you should be able to continue saying "yes" to Siri, and have infinite background processing while the car is waiting.

**Comment by ZenGuy99** [comment]
I appreciate your comment!

Yes, I have experienced Siri’s “Keep waiting?” response and, in fact, it has worked. It just felt inelegant considering how sophisticated commanding the Tesla via your app on the watch feels to the user. The public never knows the hoops programmers are jumping thru in the face of Apple defined constraints. I applaud your contribution and continued support. Super work.

Mark
Atlanta, GA
